mandatory/2-function-creation.js

```diff
@@ -4,37 +4,84 @@
     1. the user should be 18 or older
     2. the user must be logged in
 */
-function isAcceptableUser(userAge, isLoggedIn) {}
+function isAcceptableUser(userAge, isLoggedIn) {
+  return userAge >= 18 && isLoggedIn;
+}
+
+const loggedIn = true;
+const userAge = 21;
+
+if (isAcceptableUser(userAge, loggedIn)) {
+  console.log("User is acceptable.");
+} else {
+  console.log("User is not acceptable.");
+}
 
 /*
   Complete the function to apply discount percent based on how much is totalPrice in user cart.
     - When the total price is greater than 200, a 10% discount should be applied
     - When the total price is less than 200, a 5% discount should be applied
 
   The function should return the new price to be paid (e.g. if the totalPrice is 150, a 5% discount
-    is applieds and 142.5 should be returned)
+    is applied and 142.5 should be returned)
 */
 
-function applyDiscount(totalPrice) {}
+function applyDiscount(totalPrice) {
+  let discountPercent;
+  if(totalPrice > 200) {
+    discountPercent = 0.1;
+  } else {
+    discountPercent = 0.05;
+  }
+  let discountAmount = totalPrice * discountPercent;
+  let newPrice = totalPrice - discountAmount;
+  return newPrice;
+}
 
 /*
   Complete the function to print to the console the odd numbers between 1 and limit (use a while loop):
   */
-function printOddNumbers(limit) {}
+function printOddNumbers(limit) {
+  let i = 1;
+  while (i <= limit) {
+    if(i % 2 !==0) {
+      console.log(i);
+    }
+    i++;
+}
+}
 
 /*
   Complete the buyTwoGetTheCheapestFree function: if user buys two items, the cheapest item will be free!
   The function should return the price to be paid once the discount is applied.
 */
-function buyTwoGetTheCheapestFree(price1, price2) {}
+function buyTwoGetTheCheapestFree(price1, price2) {
+  if (price1 < 0 || price2 < 0) {
+    return "Invalid price for an item";
+  }
+  
+  let totalPrice = price1 + price2;
+  let discount = Math.min(price1, price2);
+  let discountedPrice = totalPrice - discount;
+  
+  return discountedPrice;
```

Comment:
I think this can be simplified with the following:

```suggestion
  return Math.max(price1, price2);
```
What do you think?